src/models/differentiable/evotree_model.jl

```diff
@@ -0,0 +1,147 @@
+using DataFrames
+using SliceMap
+using EvoTrees
+using MLJBase
+
+"""
+This type provides a basic interface to gradient-boosted tree models from the MLJ library.
+However, this is not be the final version of the interface: full support for EvoTrees has not been implemented yet and the `logits` and `probs` methods will be changed in the process of doing that if needed.
+"""
+
+"""
+    EvoTreeModel <: AbstractMLJModel
+
+Constructor for gradient-boosted decision trees from the EvoTrees.jl library. 
+"""
+struct EvoTreeModel <: AbstractDifferentiableModel
+    model::Any
+    likelihood::Symbol
+    function EvoTreeModel(model, likelihood)
+        if likelihood ∈ [:classification_binary, :classification_multi]
+            new(model, likelihood)
+        else
+            throw(
+                ArgumentError(
+                    "`type` should be in `[:classification_binary, :classification_multi]1.
+                    Support for regressors has not been implemented yet.`"
+                ),
+            )
+        end
+    end
+end
+
+"""
+Outer constructor method for EvoTreeModel.
+"""
+function EvoTreeModel(model::Any; likelihood::Symbol=:classification_binary)
+    return EvoTreeModel(model, likelihood)
+end
```

Comment:
This function looks to be quite random to me, in my opinion, it can cause more confusion from a user perspective rather than benefit them.

Reply:
The outer constructor is required because of some Julia quirks, if you look at FluxModel and DeepEnsemble, they also have the same kinds of outer constructors.

Reply:
Are you sure it's required? I commented it out, ran tests and it worked just fine.

Reply:
Okay, so it's not required, but it allows us to define an EvoTreeModel without specifying the likelihood - structs can't include default arguments. I don't think we strictly need this, but since the other classes defined by Patrick all have an outer constructor for this reason, maybe we should leave it in?

Reply:
Yes, that's a good point. Let's leave it in.